Dockerfile.default-image-builder

```diff
@@ -0,0 +1,20 @@
+ARG MAMBAFORGE_VERSION=23.11.0-0
+FROM condaforge/mambaforge:$MAMBAFORGE_VERSION as build
```

Comment:
Should we add a GA workflow to build the image?
https://github.com/flyteorg/flytekit/blob/7ddfb9b0873f33500045677b8a10942f41363048/.github/workflows/pythonpublish.yml#L76